### PR TITLE
CLOUDP-261982: atlasfederatedauth: use indexer instead of resourcewatcher

### PR DESCRIPTION
**Pull request by s-urbaniak**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by josvazg — APPROVED** [review]
LGTM, if you convince the CI as well

File: pkg/indexer/atlasfederatedauthsecret.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import (
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+const (
+	AtlasFederatedAuthBySecretsIndex = "atlasfederatedauth.spec.connectionSecret"
```
Comment (helderjs) [inline pkg/indexer/atlasfederatedauthsecret.go:11]: Similar to the other PR, dataFed copied the connection secret implementation from deployment and also relied on matching DBUsers against the project. It would really benefit an index to the project. But now I believe it should be done in a separate PR, to also deduplicate the connection secret implementation.

**Review by helderjs — APPROVED** [review]

File: pkg/indexer/atlasfederatedauthsecret.go (hunk 1 above)
Comment (s-urbaniak) [reply pkg/indexer/atlasfederatedauthsecret.go:11]: yep, agreed, this should be a separate PR.
